src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -23,31 +24,37 @@ public Program(IServiceProvider serviceProvider)
             _serviceProvider = serviceProvider;
         }
 
-        public void Main(string[] args)
+        public IHostingEngine CreateHost(IConfiguration config)
         {
-            var config = new Configuration();
-            if (File.Exists(HostingIniFile))
-            {
-                config.AddIniFile(HostingIniFile);
-            }
-            config.AddEnvironmentVariables();
-            config.AddCommandLine(args);
-
-            var engine = WebHost.CreateEngine(_serviceProvider, config);
+            var builder = new WebHostBuilder(_serviceProvider);
             var server = config.Get("server");
             if (server != null)
             {
-                engine.UseServer(server);
+                builder.UseServer(server);
             }
             var startup = config.Get("app");
```

Comment:
This code should be gone. Passing the configuration to the WebHostBuilder makes this logic obsolete as the builder should use the config system passed in as a sign to set these default.